packages/theme/components/layout/global-search/GlobalSearch.tsx

```diff
@@ -0,0 +1,276 @@
+            )}
+            tabIndex={-1}
+          >
+            <Stack direction="vertical" padding="none" gap="none">
+              {searchTerm && (
+                <Heading
+                  as="h3"
+                  size="subhead-large"
+                  id="search-results-heading"
+                  className={styles.GlobalSearch__searchResultsHeading}
+                >
+                  {searchResults.length} Results for &quot;{searchTerm}&quot;
+                </Heading>
+              )}
+              {searchResults.length > 0 ? (
+                <ul
+                  role="listbox"
+                  ref={listboxRef}
+                  id="search-results-listbox"
+                  aria-labelledby="search-results-heading"
+                  className={clsx(styles.GlobalSearch__searchResultsList)}
+                >
+                  {searchResults.map((result, index) => (
+                    <li
+                      key={`${result.title}-${index}`}
+                      className={clsx(styles.GlobalSearch__searchResultItem)}
+                      id={`search-result-${index}`}
+                      aria-selected={index === activeDescendant}
+                      role="option"
+                    >
+                      <Link
+                        className={styles.GlobalSearch__searchResultLink}
+                        href={result.url}
+                        tabIndex={-1}
+                        onClick={() => {
+                          onNavigate?.()
+                          resetSearch()
+                        }}
+                      >
+                        <Text size="200">
+                          <HighlightSearchTerm searchTerm={searchTerm}>{result.title}</HighlightSearchTerm>
+                        </Text>
+                        <Text as="p" size="100" variant="muted">
+                          <HighlightSearchTerm searchTerm={searchTerm}>{result.description}</HighlightSearchTerm>
+                        </Text>
+                      </Link>
+                    </li>
+                  ))}
+                </ul>
+              ) : (
+                <div className={styles.GlobalSearch__searchResultsEmpty}>
+                  <Text variant="muted">No results found</Text>
+                </div>
+              )}
+            </Stack>
+          </div>
+        )}
+      </div>
+    )
+  },
+)
+
+type HighlightSearchTermProps = {
+  children: React.ReactNode
+  searchTerm: string
+}
+
+const HighlightSearchTerm = ({children, searchTerm}: HighlightSearchTermProps) => {
+  if (!children || !searchTerm) {
+    return <>{children}</>
+  }
+
+  const escapedSearchTerm = searchTerm.replace(/[.*+?^${}()|[\]\\<>]/g, '\\$&')
+
+  const parts = children.toString().split(new RegExp(`(${escapedSearchTerm})`, 'gi'))
+
+  return (
+    <>
+      {parts.map((part, i) => (part.toLowerCase() === searchTerm.toLowerCase() ? <mark key={i}>{part}</mark> : part))}
+    </>
+  )
+}
```

Comment:
Could we pull this out to a separate file please? Bit buried here.

Reply:
Done 👍